src/QuokkaSimulation.hpp

```diff
@@ -321,6 +321,53 @@ template <typename problem_t> void QuokkaSimulation<problem_t>::defineComponentN
 	}
 }
 
+// initialize metadata
+template <typename problem_t> void AMRSimulation<problem_t>::initializeSimulationMetadata()
+{
+	if constexpr (Physics_Traits<problem_t>::unit_system == UnitSystem::CONSTANTS) {
+		// if unit system is CONSTANTS, the units are not well defined unless all four constants, G, k_B, c, and a_rad, are defined. However, in a hydro
+		// simulation, only k_B is defined. In a radiation-hydrodynamics simulation, only k_B, c, and a_rad are defined. Besides, CONSTANTS is only used
+		// for testing purposes, so we don't care about the units in that case. units
+		simulationMetadata_["unit_length"] = "undefined";
+		simulationMetadata_["unit_mass"] = "undefined";
+		simulationMetadata_["unit_time"] = "undefined";
+		simulationMetadata_["unit_temperature"] = "undefined";
```

Comment:
You can just set it equal to `NAN` directly. This will allow it to be read in as a floating-point number, rather than as a string, i.e.:
```
simulationMetadata_["unit_length"] = NAN;
```

Reply:
Sounds a good idea.